oshdb-api/src/main/java/org/heigit/ohsome/oshdb/api/db/OSHDBH2.java

```diff
@@ -80,29 +80,22 @@ public OSHDBH2 inMemory(boolean cache) throws SQLException {
       }
 
       Consumer<String> copyData = tablename -> {
-        try (Statement srcStmt = src.createStatement()) {
-          ResultSet rs = srcStmt.executeQuery("show columns from " + tablename);
-          List<String> columnNames = new LinkedList<>();
-          while (rs.next()) {
-            columnNames.add(rs.getString(1));
-          }
+        try (Statement srcStmt = src.createStatement();) {
+          List<String> columnNames = columnNames(tablename, srcStmt);
           String columns = columnNames.stream().collect(Collectors.joining(", "));
-          String placeholders = columnNames.stream()
-              .map(ignored -> "?")
-              .collect(Collectors.joining(", "));
-
-          PreparedStatement destStmt = this.connection.prepareStatement(
-              "insert into " + tablename + "(" + columns + ") values (" + placeholders + ")"
-          );
-          rs = srcStmt.executeQuery("select " + columns + " from " + tablename);
-          while (rs.next()) {
-            for (int i = 1; i <= columnNames.size(); i++) {
-              destStmt.setObject(i, rs.getObject(i));
+          String insertSql = insertSql(tablename, columnNames, columns);
+          String querySql = String.format("select %s from %s", columns, tablename);
+          try (var destStmt = this.connection.prepareStatement(insertSql);
+              var rs = srcStmt.executeQuery(querySql);) {
```

Comment:
Is `;` usually placed at the end of a `with` statement?
```suggestion
              var rs = srcStmt.executeQuery(querySql)) {
```